lib/trie/trie.go

```diff
@@ -355,24 +353,37 @@ func findNextKeyChild(children []*Node, startIndex byte,
 
 // Put inserts a value into the trie at the
 // key specified in little Endian format.
-func (t *Trie) Put(keyLE, value []byte) (err error) {
+func (t *Trie) Put(keyLE, value []byte, version Version) (err error) {
 	pendingDeltas := tracking.New()
 	defer func() {
 		const success = true
 		t.handleTrackedDeltas(success, pendingDeltas)
 	}()
-	return t.insertKeyLE(keyLE, value, pendingDeltas)
+	return t.insertKeyLE(keyLE, value, pendingDeltas, version)
 }
 
 func (t *Trie) insertKeyLE(keyLE, value []byte,
-	pendingDeltas DeltaRecorder) (err error) {
+	pendingDeltas DeltaRecorder, version Version) (err error) {
 	nibblesKey := codec.KeyLEToNibbles(keyLE)
 	if value == nil {
 		// Force nil value to be inserted to []byte{} since `nil` means there
 		// is no value.
 		value = []byte{}
 	}
-	root, _, _, err := t.insert(t.root, nibblesKey, value, pendingDeltas)
+
+	shouldHash := version.ShouldHashValue(value)
+	if shouldHash {
+		hashedValue := common.MustBlake2bHash(value)
+
+		// Add the original value as value node in db using the hashed value as key
+		err = t.db.Put(hashedValue.ToBytes(), value)
+		if err != nil {
+			return err
+		}
+		value = hashedValue.ToBytes()
```

Comment:
```suggestion
		// Add the original value as value node in db using the hashed value as key
		value = hashedValue.ToBytes()
		err = t.db.Put(bytes, value)
		if err != nil {
			return err
		}
```

Reply:
Where is `bytes` defined in that version? are you thinking something like this?

Reply:
I changed it this way

```suggestion
		valueKey := hashedValue.ToBytes()
		// Add the original value as value node in db using the hashed value as key
		err = t.db.Put(valueKey, value)
		if err != nil {
			return err
		}
		// Put the value key as this trie node value
		value = valueKey
```